src/zui/ZUICreatePerson/index.tsx

```diff
@@ -0,0 +1,200 @@
+                      ? tags.filter((item) => item !== value)
+                      : [...prev, value as number]
+                  );
+                } else {
+                  setPersonalInfo((prev) => {
+                    return { ...prev, [field]: value as string };
+                  });
+                }
+              }
+            }}
+            personalInfo={personalInfo}
+            tags={tags}
+          />
+        )}
+        <Box sx={{ pr: '40px' }}>
+          <Divider />
+          <Box
+            alignItems="center"
+            display="flex"
+            justifyContent="flex-end"
+            mt={2}
+          >
+            <Box>
+              <Button
+                onClick={() => {
+                  onClose();
+                  setPersonalInfo({});
+                  setTags([]);
+                }}
+                sx={{ mr: 2 }}
+                variant="text"
+              >
+                <Msg id={messageIds.createPerson.cancel} />
+              </Button>
+              <Button
+                disabled={
+                  personalInfo.first_name === undefined ||
+                  personalInfo.last_name === undefined ||
+                  checkInvalidFields(customFields, personalInfo).length !== 0
+                }
+                onClick={async (e) => {
+                  const person = await createPerson(personalInfo, tags);
+                  if (onSubmit) {
+                    onSubmit(e, person);
+                  }
+                  onClose();
+                  setPersonalInfo({});
+                  setTags([]);
+                }}
+                variant="contained"
+              >
+                {submitLabel ?? (
+                  <Msg id={messageIds.createPerson.submitLabel.default} />
+                )}
+              </Button>
+            </Box>
+          </Box>
+        </Box>
+      </Box>
+    </Dialog>
+  );
+};
+
+export default ZUICreatePerson;
+
+export const checkInvalidFields = (
+  customFields: ZetkinCustomField[],
+  personalInfo: ZetkinCreatePerson
+) => {
+  let invalidFields: string[] = [];
+
+  const customFieldURLSlugs = customFields
+    .filter((item) => item.type === 'url')
+    .map((item) => item.slug);
+
+  //email
+  if (!isEmail(personalInfo.email || '') && personalInfo.email !== undefined) {
+    invalidFields.push('email');
+  } else {
+    invalidFields = invalidFields.filter((item) => item !== 'email');
+  }
+
+  //phones
+  if (
+    !isValidPhoneNumber(personalInfo.phone || '') &&
+    personalInfo.phone !== undefined
+  ) {
+    invalidFields.push('phone');
+  } else {
+    invalidFields = invalidFields.filter((item) => item !== 'phone');
+  }
+  if (
+    !isValidPhoneNumber(personalInfo.alt_phone || '') &&
+    personalInfo.alt_phone !== undefined
+  ) {
+    invalidFields.push('alt_phone');
+  } else {
+    invalidFields = invalidFields.filter((item) => item !== 'alt_phone');
+  }
+  //urls;
+  customFieldURLSlugs.forEach((slug) => {
```

Comment:
I find the code less readable because you have this variable `customFieldUrlSlugs` that fills no other purpose than to contain a list of the fields that are URLs. But you could just loop over the fields here (instead of doing that 30 lines further up), e.g.:

```tsx
customFields.forEach((field) => {
  if (field.type == 'url') {
    const slug = field.slug;
    // Validation logic goes here
  }
});
```